### PR TITLE
Migrate to axios, small refactoring

### PR DESCRIPTION
**Pull request by MaksimShakavin**
Hello guys, 
Following our discussion in https://github.com/hepcat72/node-red-contrib-life360/issues/8 I replaced a deprecated "request" library with an axios. 
Also did a small refactoring of an error handling and how we are working with promises.
I've installed it to my home automation NR. So will keep an eye on it a couple of days and test. But I would really appreciate your testing and review

### CODE REVIEW

**Comment by hepcat72** [comment]
How has your testing gone? I have not had the opportunity to test. Still ridic busy with so many other things.

**Comment by MaksimShakavin** [comment]
So far so good. I can see connection error in the logs from time to time, but none of it crashes NR
image

File: index.js [hunk 1]
```diff
@@ -15,149 +36,72 @@ const LIFE360_CLIENT_SECRET = "cFJFcXVnYWJSZXRyZTRFc3RldGhlcnVmcmVQdW1hbUV4dWNyR
...
-        }
-        response = response.body;
-        if (!response['access_token']) throw new Error("Unauthorized");
-        const session = {
-          access_token: response['access_token'],
-          token_type: response['token_type']
+    const bodyFormData = Object.keys(form).reduce((formData, key) => {
+        formData.append(key, form[key]);
+        return formData;
+    },  new FormData())
```
Comment (brianmay) [inline index.js:52]: Wonder if this will solve #2.

**Review by brianmay — APPROVED** [review]
Reducing 483 lines down to 248 lines of code. According to github stats. So good :-)

**Comment by brianmay** [comment]
I think maybe those `"Cannot read property 'status' of undefined` messages might indicate a problem somewhere.

Not so concerned about the 503 errors, looks like these come from the server.

File: index.js (hunk 1 above)
Comment (MaksimShakavin) [reply index.js:52]: Might be, will try to test with a password, containing "+" sign.

**Comment by hepcat72** [comment]
This is great. Thanks for the contributions guys.  I haven't had real time to do anything other than briefly skim these changes, but I will get around to it eventually.  Going to go ahead and merge since I think you guys are on top of it.

**Comment by MaksimShakavin** [comment]
> I think maybe those `"Cannot read property 'status' of undefined` messages might indicate a problem somewhere.
> 
> Not so concerned about the 503 errors, looks like these come from the server.

you are right, this 'status of undefined' is coming from an error handling, when request isn't actually made(possibly because of DNS errors that I face sometimes) and request is undefined. I will fix it in the next pr

**Pull request merged**